Tests/MaterialLib/TestCapillaryPressureVanGenuchten.cpp

```diff
@@ -0,0 +1,234 @@
+        "   <residual_gas_saturation>0.05</residual_gas_saturation>"
+        "   <exponent>0.79</exponent>"
+        "   <entry_pressure>5000.0</entry_pressure>"
+        "   <maximum_capillary_pressure>1.e+20</maximum_capillary_pressure>"
+        "</property>";
+    auto const cp_ptr = createVanGenuchtenProperty(xml_pc_S, false);
+    MPL::Property const& capillary_pressure_property = *cp_ptr;
+
+    MPL::VariableArray variable_array;
+    ParameterLib::SpatialPosition const pos;
+    double const t = std::numeric_limits<double>::quiet_NaN();
+    double const dt = std::numeric_limits<double>::quiet_NaN();
+
+    std::random_device rd;
+    std::mt19937 mt(rd());
+    const double offset = std::sqrt(std::numeric_limits<double>::epsilon());
+    double const residual_liquid_saturation = 0.1;
+    double const residual_gas_saturation = 0.05;
+    std::uniform_real_distribution<double> distributor(
+        residual_liquid_saturation + offset,
+        1.0 - residual_gas_saturation - offset);
+
+    const int n = 20;
+    for (int i = 0; i <= n; ++i)
+    {
+        double const S = distributor(mt);
+        variable_array[static_cast<int>(MPL::Variable::liquid_saturation)] = S;
+
+        const double pc = capillary_pressure_property.template value<double>(
+            variable_array, pos, t, dt);
+        variable_array[static_cast<int>(MPL::Variable::capillary_pressure)] =
+            pc;
+
+        const double computed_saturation =
+            saturation_property.template value<double>(variable_array, pos, t,
+                                                       dt);
+
+        ASSERT_LE(std::fabs(S - computed_saturation), 1e-9)
+            << "for saturation " << S
+            << " and re-computed saturation via capillary pressure"
+            << computed_saturation;
+
+        const double dPcdS =
+            capillary_pressure_property.template dValue<double>(
+                variable_array,
+                MaterialPropertyLib::Variable::liquid_saturation, pos, t, dt);
+
+        const double d2PcdS2 =
+            capillary_pressure_property.template d2Value<double>(
+                variable_array,
+                MaterialPropertyLib::Variable::liquid_saturation,
+                MaterialPropertyLib::Variable::liquid_saturation, pos, t, dt);
+
+        const double S1 =
+            std::clamp(S + offset, residual_liquid_saturation + 2 * offset,
+                       1.0 - residual_gas_saturation - 2 * offset);
+        variable_array[static_cast<int>(MPL::Variable::liquid_saturation)] = S1;
+
+        if (std::fabs(S1 - S) < std::numeric_limits<double>::epsilon())
+        {
+            continue;
+        }
+
+        const double pc1 = capillary_pressure_property.template value<double>(
+            variable_array, pos, t, dt);
+        const double numerical_dPcdS = (pc1 - pc) / (S1 - S);
+        ASSERT_LE(std::fabs((dPcdS - numerical_dPcdS) / dPcdS), 1e-4)
```

Comment:
1e-4 is rather crude estimate for a derivative.

Reply:
The saturation is generated randomly. If saturation is close to its residual value, the capillary pressure and its derivative become very large (infinite). This gives the low accuracy of the numerical  derivative at S->S_r. If we fixed some saturation values for the unit test, the tolerance can be reduced to a very small value.

Reply:
Besides, the value of dpc/dS or `d^2pc/dS^2` is large too. For example
```
Expected: (std::fabs((dPcdS - numerical_dPcdS) / dPcdS)) <= (1e-6), actual: 1.03874e-06 vs 1e-06
for the analytic derivative of dPc/dS -68935.282428887251 and numeric derivative of dPc/dS.-68935.354034423828
```